docs/examples/example_simple_oosm.py

```diff
@@ -0,0 +1,278 @@
+                                  timestamp=truth[k].timestamp))
+    # store the scans into a list for plotting
+    scans_detections.append(detections)
+
+    # Scans for tracking and reordering
+    scans.append((truth[k].timestamp + timedelta(seconds=delay), detections))
+
+# Reorder the scans by their arrival time
+arrival_time_ordered = sorted(scans, key=lambda dscan: dscan[0])
+
+# %%
+# 2. Instantiate the tracking components;
+# ---------------------------------------
+# We have the scans containing the detections ordered by their arrival time.
+# It is time to prepare the tracking components, in this simple example we employ a
+# :class:`~.KalmanPredictor` and :class:`~.KalmanUpdater` to perform the tracking.
+#
+
+from stonesoup.updater.kalman import KalmanUpdater
+from stonesoup.predictor.kalman import KalmanPredictor
+predictor = KalmanPredictor(transition_model)
+updater = KalmanUpdater(measurement_model)
+
+# create the prior
+from stonesoup.types.state import GaussianState
+
+prior = GaussianState(state_vector=[0, 1, -100, 0.3],
+                      covar=np.diag([1, 1, 1, 1]),
+                      timestamp=start_time)
+
+# duplicate the prior
+prior_lag = deepcopy(prior)
+
+# %%
+# 3. Run the tracker and visualise the results.
+# ---------------------------------------------
+# We have the detections and the tracking components ready to be used.
+#
+# It is known that some detections arrived with a fixed delay, therefore we create a
+# buffer storage of fixed dimension (:math:`\ell`) where we store the detections and where we check the
+# timestamps and adjust them to their correct order. When one detection comes with an
+# earlier timestamp (:math:`\tau < t_{k}`), we shuffle the detections to keep the chain of detections.
+#
+# To show how the delayed detections impact the tracking we run a tracker with the detections
+# as they arrive to the tracker without any modification.
+
+# Load tracking components
+from stonesoup.types.hypothesis import SingleHypothesis
+from stonesoup.types.track import Track
+
+# Instantiate the empty tracks
+track = Track(prior)
+track_lag = deepcopy(track)
+
+# Load the plotter
+from stonesoup.plotter import AnimatedPlotterly
+
+# Consider the case without the algorithm
+for k in range(len(arrival_time_ordered)):  # loop over the scans
+
+    scan = arrival_time_ordered[k][1]
+
+    for detection in scan:
+        prediction = predictor.predict(prior_lag, timestamp=detection.timestamp)
+        hypothesis = SingleHypothesis(prediction, detection)
+        post = updater.update(hypothesis)
+        track_lag.append(post)
+        prior_lag = track_lag[-1]
+
+# %%
+# Visualise the tracking
+# ^^^^^^^^^^^^^^^^^^^^^^
+# Now visualise the detections and the track without the algorithm applied.
+
+plotter = AnimatedPlotterly(timesteps=timestamps)
+plotter.plot_ground_truths(truths, [0, 2])
+plotter.plot_measurements(scans_detections, [0, 2], measurements_label='Detections',
```

Comment:
See comment on line 123. `scans_detections` should be extracted in order of arrival time for continuity.